### PR TITLE
add new tests for pypi setup.py

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
Fixes:

Added New Tests for pypi setup.py

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

**Comment by Siddhant-K-code** [comment]
@majurg @pombredanne Please Review!

**Review by steven-esser — COMMENTED** [review]
@Siddhant-K-code What is the related issue these tests are for?

**Comment by Siddhant-K-code** [comment]
> @Siddhant-K-code What is the related issue these tests are for?

The issue is currently not there, but it can create a mess/issue if a `setup.py` file doesn't have a name. as similar as #2382 ( which get fixed by #2457 )

**Review by pombredanne — COMMENTED** [review]
Thanks!
Do you know if if is possible to even get a working setup.py without a version or without a name?

**Comment by Siddhant-K-code** [comment]
> Thanks!
> Do you know if if is possible to even get a working setup.py without a version or without a name?

yes, it is a good practice to do so?

**Comment by pombredanne** [comment]
> > Thanks!
> > Do you know if if is possible to even get a working setup.py without a version or without a name?
> 
> yes, it is a good practice to do so?

It is probably a bad practice, but we have to deal with it! It create an "UNKNOWN" name.

**Comment by pombredanne** [comment]
Do you mind to rebase on the latest and we can then find a proper way to handle this case?

File: tests/packagedcode/test_pypi.py [hunk 1]
```diff
@@ -224,6 +224,19 @@ def test_parse_setup_py_xmltodict(self):
...
+        packages = pypi.parse_setup_py(test_file)
+        expected_loc = self.get_test_loc('pypi/setup.py/with_name.py.expected')
+        self.check_packages(packages, expected_loc, regen=False)
+
+    def test_parse_setup_py_without_name(self):
+        test_file = self.get_test_loc('pypi/setup.py/without_name.py')
+        try:
+            pypi.parse_setup_py(test_file)
+        except AttributeError as e:
+            assert "'NoneType' object has no attribute 'to_dict'" in str(e)
```
Comment (steven-esser) [inline tests/packagedcode/test_pypi.py:238]: I would probably be nicer to have a named exception for this or some custom logged error message so we know what exactly is wrong with the setup.py pypi package (in this case, a missing name field).

**Comment by pombredanne** [comment]
This has been merged with https://github.com/nexB/scancode-toolkit/commit/accd6306871280b786c3d66b4337f04cad7b13fa after a rebase. Thank you ++
Closing now

**Pull request closed without merging**